eth2/utils/bls/src/lib.rs

```diff
@@ -6,23 +6,11 @@ mod keypair;
 mod public_key;
 mod secret_key;
 
-#[cfg(not(debug_assertions))]
```

Comment:
I did it on purpose, although I should probably have raised it with you, so apologies. 

AFAIK, the `FakeSignature` was being used everywhere (`Signature` is used) because of this line:
```
pub use crate::fake_signature::FakeSignature as Signature;
```
It meant I couldn't test the invalid signature scenario because it would always pass.
After removing the `FakeSignature` alias, I ran system-wide tests and everything ran green, so I felt confident it didn't have any repercussions.

Please let me know if you think differently or I have misread the situation.

Reply:
No worries!

The use of `Signature` vs. `FakeSignature` is controlled by those `cfg` attributes.

`#[cfg(not(debug_assertions))]` tells the compiler _"compile this if `debug_assertions` is false"_ which, in our case, loosely translates to _"compile this if we're building in `release"`_. Equivalently, `#[cfg(debug_assertions)]` says _"compile this if we're building in debug"_. (Note: checking for `debug_assertions` isn't always a fail-safe way to detect if we're running in release, but it works for our setup).

We've done this because crypto operations are slow (veery slow when building in debug) and were significantly slowing down our tests, even those tests that weren't testing crypto operations. So, our workaround is to use the fake variants when building in debug and real variants when building in release.

What this means for testing is that if you're going to run a test that requires real crypto, add the`#[cfg(not(debug_assertions))]` attribute above the test. When you want to run this test, use `$ cargo run --release`.

If you checkout our [`.travis.yml`](https://github.com/sigp/lighthouse/blob/master/.travis.yml) you can see we have to run our tests in debug _and_ in release. This is a bit annoying, but IMO less annoying that always having really slow tests.

Hopefully this is enough info, feel free to ping me if not :)